src/Arcus.Templates.AzureFunctions.ServiceBus.Topic/OrderFunction.cs

```diff
@@ -51,6 +59,45 @@ public OrderFunction(IAzureServiceBusMessageRouter messageRouter)
             AzureServiceBusMessageContext messageContext = message.GetMessageContext(_jobId);
             MessageCorrelationInfo correlationInfo = message.GetCorrelationInfo();
             await _messageRouter.RouteMessageAsync(message, messageContext, correlationInfo, cancellationToken);
+        } 
+#endif
+#if Isolated
+        /// <summary>
+        /// Process an Azure Service Bus topic <paramref name="messageBody"/> as an <see cref="Order"/>.
+        /// </summary>
+        /// <param name="messageBody">The incoming message on the Azure Service Bus topic, representing an <see cref="Order"/>.</param>
+        /// <param name="context">The execution context for this Azure Functions instance.</param>
+        [Function("order-processing")]
+        public async Task Run(
+            [ServiceBusTrigger("order-topic", "order-subscription", Connection = "ServiceBusConnectionString")] byte[] messageBody,
+            FunctionContext context)
+        {
+            ServiceBusReceivedMessage message = ConvertToServiceBusMessage(messageBody, context);
+            var logger = context.GetLogger<ILogger<OrderFunction>>();
+            logger.LogInformation($"C# ServiceBus topic trigger function processed message: {message.MessageId}");
+
+            AzureServiceBusMessageContext messageContext = message.GetMessageContext(_jobId);
+            MessageCorrelationInfo correlationInfo = message.GetCorrelationInfo();
+            await _messageRouter.RouteMessageAsync(message, messageContext, correlationInfo, CancellationToken.None);
+        }
+
+        private static ServiceBusReceivedMessage ConvertToServiceBusMessage(byte[] messageBody, FunctionContext context)
+        {
+            var applicationProperties = new Dictionary<string, object>();
+            if (context.BindingContext.BindingData.TryGetValue("ApplicationProperties", out object applicationPropertiesObj))
+            {
+                var json = applicationPropertiesObj.ToString();
+                applicationProperties = JsonSerializer.Deserialize<Dictionary<string, object>>(json);
+            }
+
+            var message = ServiceBusModelFactory.ServiceBusReceivedMessage(
+                body: BinaryData.FromBytes(messageBody),
+                messageId: context.BindingContext.BindingData["MessageId"]?.ToString(),
+                correlationId: context.BindingContext.BindingData["CorrelationId"]?.ToString(),
+                properties: applicationProperties);
+
+            return message;
         }
```

Comment:
Message-specific metadata is not yet supported for isolated functions. Though, they are already in the binding context.
https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-service-bus-trigger?tabs=isolated-process%2Cextensionv5&pivots=programming-language-csharp#message-metadata